pythonFiles/vscode_pytest/__init__.py

```diff
@@ -0,0 +1,340 @@
+        {
+            "name": session.name,
+            "path": os.fspath(session.path),
+            "type_": "folder",
+            "children": [],
+            "id_": os.fspath(session.path),
+        }
+    )
+
+
+def create_class_node(class_module: pytest.Class) -> TestNode:
+    """
+    Creates a class node from a pytest class object.
+
+    Keyword arguments:
+    class_module -- the pytest object representing a class module.
+    """
+    return TestNode(
+        {
+            "name": class_module.name,
+            "path": os.fspath(class_module.path),
+            "type_": "class",
+            "children": [],
+            "id_": class_module.nodeid,
+        }
+    )
+
+
+def create_file_node(file_module: Node) -> TestNode:
+    """
+    Creates a file node from a pytest file module.
+
+    Keyword arguments:
+    file_module -- the pytest file module.
+    """
+    return TestNode(
+        {
+            "name": file_module.path.name,
+            "path": os.fspath(file_module.path),
+            "type_": "file",
+            "id_": os.fspath(file_module.path),
+            "children": [],
+        }
+    )
+
+
+def create_doc_file_node(file_module: pytest.Module) -> TestNode:
+    """
+    Creates a doc file node from a pytest doc test file module.
+
+    Keyword arguments:
+    file_module -- the module for the doc test file.
+    """
+    return TestNode(
+        {
+            "name": file_module.path.name,
+            "path": os.fspath(file_module.path),
+            "type_": "doc_file",
+            "id_": os.fspath(file_module.path),
+            "children": [],
+        }
+    )
+
+
+def create_folder_node(folderName: str, path_iterator: pathlib.Path) -> TestNode:
+    """
+    Creates a folder node from a pytest folder name and its path.
+
+    Keyword arguments:
+    folderName -- the name of the folder.
+    path_iterator -- the path of the folder.
+    """
+    return TestNode(
+        {
+            "name": folderName,
+            "path": os.fspath(path_iterator),
+            "type_": "folder",
+            "id_": os.fspath(path_iterator),
+            "children": [],
+        }
+    )
+
+
+class PayloadDict(Dict):
+    """
+    A dictionary that is used to send a post request to the server.
+    """
+
+    cwd: str
+    status: Literal["success", "error"]
+    tests: Optional[TestNode]
+    errors: Optional[List[str]]
+
+
+def post_response(cwd: str, session_node: TestNode, errors: List[str]) -> None:
+    """
+    Sends a post request to the server.
+    Keyword arguments:
+    cwd -- the current working directory.
+    session_node -- the session node, which is the top of the testing tree.
+    errors -- a list of errors that occurred during test collection.
+    """
+    if errors:
+        payload = PayloadDict(
+            {"cwd": cwd, "status": "error", "tests": session_node, "errors": errors}
+        )
+    else:
+        payload = PayloadDict({"cwd": cwd, "status": "success", "tests": session_node})
```

Comment:
```suggestion
    payload: PayloadDict = {"cwd": cwd, "tests": session_node, "status": "success" if not errors else "error"}
    if errors:
        payload["errors"] = errors
```

If you accepted a list for `"errors"` no matter what the status, then you could use `"errors": errors or []` and drop the `if`.